test/transaction.ts

```diff
@@ -278,7 +294,7 @@ describe('Transaction', () => {
 
         const {gaxOpts, reqOpts} = REQUEST_STREAM.lastCall.args[0];
 
-        assert.strictEqual(gaxOpts, fakeOptions);
+        assert.deepStrictEqual(gaxOpts, fakeOptions);
```

Comment:
```suggestion
        assert.strictEqual(config.gaxOpts, fakeOptions);
```

If after making that change, the test fails, then we would want to keep `deepStrictEqual`, but also set a property on `fakeOptions`.